File: Week 07/id_437/191.位-1-的个数.js
/*
 * @lc app=leetcode.cn id=191 lang=javascript
 *
 * [191] 位1的个数
 */

// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
    let bits = 0;
    let mask = 1;
    for (let i = 0; i < 32; i++) {
      if ((n & mask) !== 0) {
        bits++;
      }
      mask <<= 1;
    }
    return bits;
};
// @lc code=end


// @lc code=start
/**
 * @param {number} n - a positive integer
 * @return {number}
 */
var hammingWeight = function(n) {
  let sum = 0;
  while (n !== 0) {
    sum++
    n &= (n - 1);
  }
  return sum;
};
// @lc code=end

File: Week 07/id_477/191.位-1-的个数.go
package leetcode

/*
 * @lc app=leetcode.cn id=191 lang=golang
 *
 * [191] 位1的个数
 */

// @lc code=start
func hammingWeight(num uint32) int {
	var count int
	for num > 0 {
		count++
		num = num & (num - 1)
	}
	return count
}

// @lc code=end
